datafusion/sql/src/unparser/expr.rs

```diff
@@ -411,9 +411,34 @@ impl Unparser<'_> {
             Expr::Wildcard { qualifier: _ } => {
                 not_impl_err!("Unsupported Expr conversion: {expr:?}")
             }
-            Expr::GroupingSet(_) => {
-                not_impl_err!("Unsupported Expr conversion: {expr:?}")
-            }
+            Expr::GroupingSet(grouping_set) => match grouping_set {
+                GroupingSet::GroupingSets(grouping_sets) => {
+                    let expr_ast_sets = grouping_sets
+                        .iter()
+                        .map(|set| {
+                            set.iter()
+                                .map(|e| self.expr_to_sql(e))
+                                .collect::<Result<Vec<_>>>()
+                        })
+                        .collect::<Result<Vec<_>>>()?;
+
+                    Ok(ast::Expr::GroupingSets(expr_ast_sets))
+                }
+                GroupingSet::Cube(cube) => {
+                    let expr_ast_sets = cube
+                        .iter()
+                        .map(|e| vec![self.expr_to_sql(e).unwrap()])
+                        .collect::<Vec<_>>();
```

Comment:
I think it is important to not call `unwrap` here as it will panic

You can collect the error with as you do above for grouping sets

```suggestion
                        .map(|e| vec![self.expr_to_sql(e)])
                        .collect::<Result<Vec<_>>>()?;
```

(the same comment applies for `GroupingSet::Rollup`)

Reply:
My bad... done.